util/types/time.go

```diff
@@ -2043,6 +2043,49 @@ var dateFormatParserTable = map[string]dateFormatParser{
 	// "%y": yearTwoDigits,         // Year, numeric (two digits)
 }
 
+// GetFormatType checks the type(Duration, Date or Datetime) of a format string.
+func GetFormatType(format string) (isDuration, isDate bool) {
+	durationTokens := map[string]struct{}{
```

Comment:
can we put these two `map`s outside this function ?

Reply:
We only use these maps in this function,
I think we can move them outside if necessary later.